rule/max-public-structs.go

```diff
@@ -19,20 +19,22 @@ const defaultMaxPublicStructs = 5
 func (r *MaxPublicStructsRule) configure(arguments lint.Arguments) {
 	r.Lock()
 	defer r.Unlock()
-	if r.max < 1 {
-		if len(arguments) < 1 {
-			r.max = defaultMaxPublicStructs
-			return
-		}
+	if r.max == 0 {
+		return // already configured
+	}
 
-		checkNumberOfArguments(1, arguments, r.Name())
+	if len(arguments) < 1 {
+		r.max = defaultMaxPublicStructs
+		return
+	}
 
-		max, ok := arguments[0].(int64) // Alt. non panicking version
-		if !ok {
-			panic(`invalid value passed as argument number to the "max-public-structs" rule`)
-		}
-		r.max = max
+	checkNumberOfArguments(1, arguments, r.Name())
+
+	max, ok := arguments[0].(int64) // Alt. non panicking version
```

Comment:
Note, max here will override a built-in.

Reply:
right, I've created this branch before the last merged PR #1052, I will rebase once I've refactored all rules